client/test/explorer/blockchainExplorer.test.ts

```diff
@@ -30,273 +41,523 @@ describe('BlockchainExplorer', () => {
+                mySandBox.stub(fabricClientConnection, 'getAllPeerNames').resolves(['peerOne', 'peerTwo']);
+                const getAllChannelsForPeerStub = mySandBox.stub(fabricClientConnection, 'getAllChannelsForPeer');
+                getAllChannelsForPeerStub.onFirstCall().resolves(['channelOne', 'channelTwo']);
+                getAllChannelsForPeerStub.onSecondCall().resolves(['channelTwo']);
+
+                const installedChaincodeStub = mySandBox.stub(fabricClientConnection, 'getInstalledChaincode');
+
+                const installedChaincodeMapOne: Map<string, Array<string>> = new Map<string, Array<string>>();
+                installedChaincodeMapOne.set('sample-car-network', ['1.0', '1.2']);
+                installedChaincodeMapOne.set('sample-food-network', ['0.6']);
+
+                installedChaincodeStub.withArgs('peerOne').returns(installedChaincodeMapOne);
+
+                const installedChaincodeMapTwo: Map<string, Array<string>> = new Map<string, Array<string>>();
+                installedChaincodeMapTwo.set('biscuit-network', ['0.7']);
+                installedChaincodeStub.withArgs('peerTwo').returns(installedChaincodeMapTwo);
+
+                const instantiatedChaincodeStub = mySandBox.stub(fabricClientConnection, 'getInstantiatedChaincode');
+                instantiatedChaincodeStub.withArgs('channelOne').resolves([{name: 'biscuit-network', version: '0.7'}]);
+                instantiatedChaincodeStub.withArgs('channelTwo').resolves([{name: 'cake-network', version: '0.10'}]);
+
+                blockchainNetworkExplorerProvider = myExtension.getBlockchainNetworkExplorerProvider();
+                blockchainNetworkExplorerProvider['connection'] = fabricClientConnection;
+
+                allChildren = await blockchainNetworkExplorerProvider.getChildren();
             });
 
-            connections.push({
-                name: 'myConnectionA',
-                connectionProfilePath: path.join(rootPath, '../../test/data/connectionTwo/connection.json'),
-                identities: [{
-                    certificatePath: path.join(rootPath, '../../test/data/connectionTwo/credentials/certificate'),
-                    privateKeyPath: path.join(rootPath, '../../test/data/connectionTwo/credentials/privateKey')
-                }]
+            afterEach(() => {
+                mySandBox.restore();
             });
 
-            connections.push({
-                name: 'myConnectionA',
-                connectionProfilePath: path.join(rootPath, '../../test/data/connectionTwo/connection.json'),
-                identities: [{
-                    certificatePath: path.join(rootPath, '../../test/data/connectionTwo/credentials/certificate'),
-                    privateKeyPath: path.join(rootPath, '../../test/data/connectionTwo/credentials/privateKey')
-                }]
+            it('should create a connected tree if there is a connection', async () => {
```

Comment:
Where is the code being called in this test?

Reply:
the code is being called in the before each